mdoc/src/main/scala-3/mdoc/internal/markdown/MarkdownCompiler.scala

```diff
@@ -50,26 +53,28 @@ class MarkdownCompiler(
     target: AbstractFile = new VirtualDirectory("(memory)")
 ) {
 
-  private def newDriver: InteractiveDriver = {
+  private def newContext: FreshContext = {
     val defaultFlags =
       List("-color:never", "-unchecked", "-deprecation", "-Ximport-suggestion-timeout", "0")
     val options = scalacOptions.split("\\s+").toList
     val settings =
       options ::: defaultFlags ::: "-classpath" :: classpath :: Nil
-    new InteractiveDriver(settings)
+    val driver = new InteractiveDriver(settings.distinct)
+    driver.currentCtx.fresh.setReporter(new CollectionReporter)
   }
-  private var driver = newDriver
+
+  private var context = newContext
```

Comment:
After tinkering with failing warning suites, I found that Driver doesn't seem to be as important as Context

This change started making those tests pass (warnings properly surfaced)